### PR TITLE
Fix BertVectorizer for long sequences and add tests

### LINKED ISSUE #173: Bert vectorizer count of max tokens is wrong
https://github.com/wellcometrust/WellcomeML/blob/16754b0a9806b29b942a4ac67ba9b67253ce97fc/wellcomeml/ml/bert_vectorizer.py#L43

This lines tries to ensure that a max of 512 tokens are passed to BERT but tokens are not number of characters but number of words or subwords to be exact. The most accurate way to count them would be to tokenize using the bert tokenizer but we could also approximate by `x.split()`

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Fixes #173 

The fix is slightly different than the one suggested in the issue. Let me know what you think.

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by nsorros** [comment]
Interestingly the test that fails succeeds when i run it, need to investigate this a bit

**Comment by nsorros** [comment]
@aCampello @lizgzil

**Comment by nsorros** [comment]
@lizgzil @aCampello this is a relatively small PR

File: wellcomeml/ml/bert_vectorizer.py [hunk 1]
```diff
@@ -40,14 +40,15 @@ def load_transformed(cls, path):
...
     def bert_embedding(self, x):
-        # Max sequence length is 512 for BERT
-        if len(x) > 512:
-            embedded_a = self.bert_embedding(x[:512])
-            embedded_b = self.bert_embedding(x[512:])
+        tokenized_x = self.tokenizer.tokenize(x)
+
+        # Max sequence length is 512 for BERT. 510 without CLS and SEP
+        if len(tokenized_x) > 510:
+            embedded_a = self.bert_embedding(" ".join(tokenized_x[:510]))
```
Comment (aCampello) [inline wellcomeml/ml/bert_vectorizer.py:47]: I see, we are approximating by joining them with " ". It might miss some edge cases, but probably ok!

**Review by aCampello — APPROVED** [review]

**Comment by aCampello** [comment]
LGTM

File: wellcomeml/ml/bert_vectorizer.py (hunk 1 above)
Comment (nsorros) [reply wellcomeml/ml/bert_vectorizer.py:47]: whatever the reason for splitting them before, space should make them split again is my theory so its only safe to add spaces there

**Pull request merged**